src/main/java/com/wavefront/sdk/direct/ingestion/WavefrontDirectIngestionClient.java

```diff
@@ -340,36 +350,53 @@ public void run() {
-        pointsDropped, pointReportErrors);
+    internalFlush(metricsBuffer, Constants.WAVEFRONT_METRIC_FORMAT, "points", "points",
+        pointsDropped, pointReportErrors, null);
     internalFlush(histogramsBuffer, Constants.WAVEFRONT_HISTOGRAM_FORMAT, "histograms",
-        histogramsDropped, histogramReportErrors);
-    internalFlush(tracingSpansBuffer, Constants.WAVEFRONT_TRACING_SPAN_FORMAT, "spans",
-        spansDropped, spanReportErrors);
-    internalFlush(spanLogsBuffer, Constants.WAVEFRONT_SPAN_LOG_FORMAT, "span_logs",
-        spanLogsDropped, spanLogReportErrors);
+        "histograms", histogramsDropped, histogramReportErrors, histogramsDisabled);
+    internalFlush(tracingSpansBuffer, Constants.WAVEFRONT_TRACING_SPAN_FORMAT, "spans", "spans",
+        spansDropped, spanReportErrors, spansDisabled);
+    internalFlush(spanLogsBuffer, Constants.WAVEFRONT_SPAN_LOG_FORMAT, "span_logs", "span logs",
+        spanLogsDropped, spanLogReportErrors, spanLogsDisabled);
   }
 
-  private void internalFlush(LinkedBlockingQueue<String> buffer, String format, String entityPrefix,
-                             WavefrontSdkCounter dropped, WavefrontSdkCounter reportErrors)
+  private void internalFlush(LinkedBlockingQueue<String> buffer, String format,
+                             String entityPrefix, String entityType,
+                             WavefrontSdkCounter dropped, WavefrontSdkCounter reportErrors,
+                             @Nullable AtomicBoolean featureDisabled)
       throws IOException {
     List<List<String>> batch = getBatch(buffer, batchSize, messageSizeBytes, dropped);
     for (int i = 0; i < batch.size(); i++) {
       List<String> items = batch.get(i);
+      if (featureDisabled != null && featureDisabled.get()) {
+        dropped.inc(items.size());
+        continue;
+      }
       try (InputStream is = itemsToStream(items)) {
         int statusCode = directService.report(format, is);
         sdkMetricsRegistry.newCounter(entityPrefix + ".report." + statusCode).inc();
         if (400 <= statusCode && statusCode <= 599) {
-          logger.log(Level.WARNING, "Error reporting points, respStatus=" + statusCode);
-          int numAddedBackToBuffer = 0;
-          for (String item : items) {
-            if (buffer.offer(item)) {
-              numAddedBackToBuffer++;
-            } else {
-              dropped.inc(items.size() - numAddedBackToBuffer);
-              logger.log(Level.WARNING, "Buffer full, dropping attempted points");
+          switch (statusCode) {
+            case 401:
+            case 403:
+              if (featureDisabled == null) {
+                logger.log(Level.WARNING,
+                    "Permission error sending " + entityType + " to Wavefront (HTTP " +
+                        statusCode + ")");
+                requeue(buffer, items, dropped, entityType);
+              } else {
+                logger.log(Level.SEVERE,
```

Comment:
if it's possible to still print this message every 5 seconds (but not attempt to send data), meaning as long as we are receiving a data type but not flushing it because it's disabled on the server, let's log a message up to once every 5 seconds

meaning if we stopped receiving span logs from the app for that 5s interval, we would stop printing the message

Reply:
updated to print a similar message if `featureDisabled` flag is up, and print it at most once every 5 sec